pkg/controllers/clusteroperator/clusteroperator_controller_test.go

```diff
@@ -0,0 +1,203 @@
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorProgressing)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorDegraded)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorUpgradeable)), HaveField("Status", Equal(configv1.ConditionTrue)))),
+				)), "should match the expected ClusterOperator status conditions")
+		})
+
+		It("should update the ClusterOperator status version to the desired one", func() {
+			Eventually(komega.Object(configv1resourcebuilder.ClusterOperator().WithName(controllers.ClusterOperatorName).Build())).
+				Should(HaveField("Status.Versions", ContainElement(SatisfyAll(
+					HaveField("Name", Equal("operator")),
+					HaveField("Version", Equal(desiredOperatorReleaseVersion)),
+				))), "should match the expected ClusterOperator status versions")
+		})
+
+		It("should update the ClusterOperator status version to the desired one when an incorrect one is present", func() {
+			By("Setting the ClusterOperator status version to an incorrect one")
+			patchBase := client.MergeFrom(capiClusterOperator.DeepCopy())
+			capiClusterOperator.Status.Versions = []configv1.OperandVersion{{Name: "operator", Version: "incorrect"}}
+			Expect(cl.Status().Patch(ctx, capiClusterOperator, patchBase)).To(Succeed())
+
+			By("Checking the conditions are as expected")
+			Eventually(komega.Object(configv1resourcebuilder.ClusterOperator().WithName(controllers.ClusterOperatorName).Build())).
+				Should(HaveField("Status.Versions", ContainElement(SatisfyAll(
+					HaveField("Name", Equal("operator")),
+					HaveField("Version", Equal(desiredOperatorReleaseVersion)),
+				))))
+		})
+	})
+})
+
+func startManager(isUnsupportedPlatform bool) (context.CancelFunc, chan struct{}) {
+	mgrCtx, mgrCancel := context.WithCancel(context.Background())
+	mgrDone := make(chan struct{})
+
+	By("Setting up a manager and controller")
+
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: testScheme,
+		Controller: config.Controller{
+			SkipNameValidation: ptr.To(true),
+		},
+	})
+	Expect(err).ToNot(HaveOccurred(), "Manager should be able to be created")
+
+	r := &ClusterOperatorController{
+		ClusterOperatorStatusClient: operatorstatus.ClusterOperatorStatusClient{Client: cl, ReleaseVersion: desiredOperatorReleaseVersion},
+		IsUnsupportedPlatform:       isUnsupportedPlatform,
+	}
+	Expect(r.SetupWithManager(mgr)).To(Succeed(), "Reconciler should be able to setup with manager")
+
+	By("Starting the manager")
+
+	go func() {
+		defer GinkgoRecover()
+		defer close(mgrDone)
+
+		Expect((mgr).Start(mgrCtx)).To(Succeed())
+	}()
+
+	return mgrCancel, mgrDone
+}
+
+func stopManager() {
+	By("Stopping the manager")
+	mgrCancel()
+	// Wait for the mgrDone to be closed, which will happen once the mgr has stopped.
+	<-mgrDone
+
+	Eventually(mgrDone).Should(BeClosed())
```

Comment:
Nit, I think these will do the same thing no? So, do we need line 200?

Reply:
Yeah you are right, good point. Updated.